GVFS/GVFS.FunctionalTests/Tools/GVFSFunctionalTestEnlistment.cs

```diff
@@ -128,12 +131,12 @@ public string GetObjectRoot(FileSystemRunner fileSystem)
 
             this.LocalCacheRoot.ShouldBeADirectory(fileSystem).WithFiles().ShouldNotContain(f => !allowedFileNames.Contains(f.Name));
 
-            DirectoryInfo[] directories = this.LocalCacheRoot.ShouldBeADirectory(fileSystem).WithDirectories().ToArray();
-            directories.Length.ShouldEqual(
-                1,
-                this.LocalCacheRoot + " is expected to have only one folder. Actual folders: " + string.Join<DirectoryInfo>(",", directories));
+            string[] objectRootEntries = File.ReadAllLines(mappingFile).Where(x => x.IndexOf(this.RepoUrl, StringComparison.OrdinalIgnoreCase) >= 0).ToArray();
+            objectRootEntries.Length.ShouldEqual(1, $"Should be only one entry for repo url: {this.RepoUrl}");
+            objectRootEntries[0].Substring(0, 2).ShouldEqual("A ", $"Invalid mapping entry for repo: {objectRootEntries[0]}");
+            JObject rootEntryJson = JObject.Parse(objectRootEntries[0].Substring(2));
 
-            return Path.Combine(directories[0].FullName, "gitObjects");
+            return Path.Combine(this.LocalCacheRoot, rootEntryJson.GetValue("Value").ToString(), "gitObjects");

```

Comment:
Is there anything about `rootEntryJson.GetValue("Value").ToString()` we should validate before calling `Path.Combine`?